hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop-titan-provider/src/main/java/org/hawkular/inventory/impl/tinkerpop/provider/TitanProvider.java

```diff
@@ -248,4 +249,17 @@ public RuntimeException translateException(RuntimeException inputException, Cano
         }
         return inputException;
     }
+
+    @Override public boolean requiresRollbackAfterFailure(Throwable t) {
+        while (t != null) {
+            if (t instanceof TitanException
+                    && "Could not commit transaction due to exception during persistence".equals(t.getMessage())) {
+                return false;
+            }
+
+            t = t.getCause();
+        }
```

Comment:
Also that the exception we're checking there is not the root cause unfortunately...